# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1385537400 -3600
#      Wed Nov 27 08:30:00 2013 +0100
# Branch stable
# Node ID f1d791ffeeb44bcedeedb15ed7b7c2299da8dd19
# Parent  50bce63dbb04d4c2045dae26f2a946382d283467
install translators before welcome wizard but after loading settings

* octave-gui.cc (octave_start_gui): if first run: install translators, show
  welcome wizard, and load settings; otherwise: load settings and install
  translators

diff --git a/libgui/src/octave-gui.cc b/libgui/src/octave-gui.cc
--- a/libgui/src/octave-gui.cc
+++ b/libgui/src/octave-gui.cc
@@ -122,39 +122,45 @@ octave_start_gui (int argc, char *argv[]
 {
   octave_thread_manager::block_interrupt_signal ();
 
   qInstallMsgHandler (message_handler);
 
   if (start_gui)
     {
       QApplication application (argc, argv);
+      QTranslator gui_tr, qt_tr, qsci_tr;
 
       // Set the codec for all strings (before wizard)
       QTextCodec::setCodecForCStrings (QTextCodec::codecForName ("UTF-8"));
 
-      // install translators for the gui and qt text (before wizard)
-      QTranslator gui_tr, qt_tr, qsci_tr;
-
-      resource_manager::config_translators (&qt_tr, &qsci_tr, &gui_tr);
-
-      application.installTranslator (&qt_tr);
-      application.installTranslator (&qsci_tr);
-      application.installTranslator (&gui_tr);
-
       // show wizard if this is the first run
       if (resource_manager::is_first_run ())
         {
+          resource_manager::config_translators (&qt_tr, &qsci_tr, &gui_tr); // before wizard
+          application.installTranslator (&qt_tr);
+          application.installTranslator (&qsci_tr);
+          application.installTranslator (&gui_tr);
+
           welcome_wizard welcomeWizard;
 
           if (welcomeWizard.exec () == QDialog::Rejected)
             exit (1);
-        }
 
-      resource_manager::reload_settings ();
+          resource_manager::reload_settings ();  // install settings file
+        }
+      else
+        {
+          resource_manager::reload_settings ();  // get settings file
+
+          resource_manager::config_translators (&qt_tr, &qsci_tr, &gui_tr); // after settings
+          application.installTranslator (&qt_tr);
+          application.installTranslator (&qsci_tr);
+          application.installTranslator (&gui_tr);
+        }
 
       // update network-settings
       resource_manager::update_network_settings ();
 
 #if ! defined (__WIN32__) || defined (__CYGWIN__)
       // If we were started from a launcher, TERM might not be
       // defined, but we provide a terminal with xterm
       // capabilities.
